carta/image.py

```diff
@@ -672,6 +672,224 @@ def hide_contours(self):
         """Hide the contours."""
         self.set_contours_visible(False)
 
+    # VECTOR OVERLAY
+
+    @validate(NoneOr(Constant(VectorOverlaySource)), NoneOr(Constant(VectorOverlaySource)), NoneOr(Boolean()), NoneOr(Number()), NoneOr(Boolean()), NoneOr(Boolean()), NoneOr(Number()), NoneOr(Boolean()), NoneOr(Number()), NoneOr(Number()))
+    def configure_vector_overlay(self, angular_source=None, intensity_source=None, pixel_averaging_enabled=None, pixel_averaging=None, fractional_intensity=None, threshold_enabled=None, threshold=None, debiasing=None, q_error=None, u_error=None):
+        """Configure vector overlay.
+
```

Comment:
In all these parameter descriptions we need to make a distinction between the *default value* -- which is always `None` and always results in the current frontend value being preserved, and the *initial value* set by the frontend, which is determined by the store constructor, and which may additionally be configurable in the frontend preferences.

This docstring needs to make sense whether someone is calling this function for the first time (when all the frontend settings have their initial values) or if they are calling it a second time, after changing some of the values.

So we have two options here -- either consistently not mention the initial values (the minimalist approach), or fill them in for each of the parameters (more helpful given that we don't currently have documentation for the frontend functions). Either way, I think we need to add a sentence to the docstring here:

``All parameters are optional. For each option that is not provided, the value currently set in the frontend will be preserved.``

If we do want to fill in all the initial values, we should then also add ``Initial frontend settings are noted below.``

Then for each of the parameters we should put something like `Initially set to X.` or `Initially disabled.`, and add `This default is configurable in the frontend preferences.` if that value is read from the prefs.

I will give more specific examples for the parameters in this function later today.

Reply:
Updated suggestion for the additional paragraphs:
```
All parameters are optional. For each option that is not provided, the value currently set in the frontend will be preserved. Initial frontend settings are noted below.

We deduce some boolean options. For example, providing an explicit pixel averaging width with the **pixel_averaging** parameter will automatically enable pixel averaging unless **pixel_averaging_enabled** is also explicitly set to ``False``. To disable pixel averaging, explicitly set **pixel_averaging_enabled** to ``False``.
```